docs/cedarling/cedarling-jwt-validation.md

```diff
@@ -1,71 +1,191 @@
 
 - `exp` (expiration)
 - `nbf` (not before)
 
-If the Cedarling property `CEDARLING_ID_TOKEN_TRUST_MODE` is `Strict`, the Cedarling will:
+### Custom Required Claims
 
-- Discard `id_token` if the `aud` claim does not contain the `client_id` of the access token
-- Discard Userinfo tokens that are not associated with a `sub` claim from the `id_token`
+You can specify additional required claims in your token metadata configuration:
 
+```json
+{
+  // ... other fields have been omitted for brevity
+  "trusted_issuers": {
+    "my_trusted_issuer_id": {
+      "name": "my trusted issuer",
+      "description": "an IDP that i trust",
+      "openid_configuration_endpoint": "https://test.issuer.com/.well-known/openid-configuration",
+      "token_metadata": { 
+        "access_token": {
+          "entity_type_name": "Jans::Access_token",
+          "required_claims": ["exp", "client_id"]
+        }
+      }
+    }
+  }
+}
+```
 
-### JWT Status Validation
+The above configuration means that any `access_token` must contain both the `exp` and `client_id` claims, or it will be rejected. Additionally, *registered claims*[^registered-claims] like the `exp` will also be validated according to RFC 7519 standards.
 
-The Cedarling can also check for JWT revocation if you have the `CEDARLING_JWT_STATUS_VALIDATION` property `Enabled`. The Cedarling checks the status bit of the Status Token JWT, as described in the [OAuth Status Lists](https://datatracker.ietf.org/doc/draft-ietf-oauth-status-list/) draft. Token status enforcement mitigates account takeover by enabling immediate revocation of all tokens issued to an attacker. 
+[^registered-claims]: Registered claims like `exp`, `iat`, `iss`, etc., are defined in [`RFC 7519, Section 4.1`](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1)
 
 ## ID Token Trust Mode
 
-The level of validation for the ID Token JWT can be set to either `None` or `Strict`.
+Cedarling supports an optional strict trust mode for validating relationships between different token types—primarily ID tokens, Access tokens, and Userinfo tokens.
+
+This behavior is controlled via the `CEDARLING_ID_TOKEN_TRUST_MODE` [bootstrap property](#related-bootstrap-properties).
+
+> **Implementation Status:** Currently, only `strict` and `never` modes are implemented. Additional modes (`always`, `ifpresent`) are defined in the enum but not yet supported in the validation logic.
 
-### `None` Mode
+### `strict` Mode
 
-Setting the validation level to `None` will not check for the conditions outlined in [Strict Mode](#strict-mode).
+If `CEDARLING_ID_TOKEN_TRUST_MODE` is set to `strict`, Cedarling will enforce the following:
 
-### `Strict` Mode
+1. The `id_token`'s `aud` (Audience) must match the `access_token`'s `client_id`;
+2. If a `userinfo_token` is also provided:
+  - Its `sub` (Subject) must match that of the `id_token`
+  - Its `aud` must also match the `access_token`'s `client_id`.
 
```

Comment:
<img width="722" height="161" alt="image" src="https://github.com/user-attachments/assets/d0d2a950-56c8-46f4-bc63-28cd9c7d8134" />

Unnumbered list did not render correctly